plugins/stub_resolver/stub_resolver.ml

```diff
@@ -2,21 +2,30 @@ open Core_kernel[@@warning "-D"]
 open Bap.Std
 open Bap_core_theory
 open Bap_knowledge
+open Graphlib.Std
+open Regular.Std
 
 include Self ()
 let package = "bap"
 
 open KB.Syntax
 
-type groups = int Tid.Map.t
-type names = String.Set.t Int.Map.t
+module Regular_string = struct
+  type t = string
+  include Regular.Make(struct
+      include String
+      let module_name = Some "String"
+      let version = "2.6.0"
+    end)
+end
+
+module G = Graphlib.Make(Regular_string)(Unit)
 
 type state = {
-  groups : groups;
-  names  : names;
-  next   : int;
-  stubs  : Tid.Set.t;
-  units  : Theory.Unit.t Tid.Map.t;
+  graph : G.t;
+  names : (tid, string) Bap_relation.t;
```

Comment:
It's confusing why we are using Bap_relation here if we are just (ab)using it as a plain mapping. Can you please, switch to regular `string list TId.Map.t` here?